rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/AbstractRepositoryGraphLike.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j.impl;
+
+import java.util.Optional;
+
+import org.apache.commons.rdf.api.BlankNodeOrIRI;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.RDFTerm;
+import org.apache.commons.rdf.api.TripleLike;
+import org.apache.commons.rdf.rdf4j.RDF4JGraphLike;
+import org.apache.commons.rdf.rdf4j.RDF4JTermFactory;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+
+public abstract class AbstractRepositoryGraphLike<T extends TripleLike<BlankNodeOrIRI, IRI, RDFTerm>>
+		implements RDF4JGraphLike<T> {
+
+	protected Repository repository;
+	protected boolean includeInferred;
+	protected boolean shouldWeShutdown = false;
+	protected RDF4JTermFactory rdf4jTermFactory;
+
+	public AbstractRepositoryGraphLike(Repository repository) {
+		this(repository, false);
+	}
+
+	public AbstractRepositoryGraphLike(Repository repository, boolean includeInferred) {
+		this.repository = repository;
+		this.includeInferred = includeInferred;
+		if (!repository.isInitialized()) {
```

Comment:
Not sure what's the difference from Commons RDF being an "RDF4J user" or the outer Commons RDF caller being a "RDF4J user" unless the caller is running `repository.initialize()` in a separate thread at the same time as passing it to RDF4JTermFactory (which sounds like asking for trouble).

If I remove this I think we can no longer do `repository.shutdown()` from `Graph.close()` - e.g. you can no longer do the `new BufferedInputStream(new InputStream)` pattern:

```
Triple t;
Sail sail = new NativeStore(file);
try (Graph g = RDF4JTermFactory.asGraph(new SailRepository(sail)) {
  g.add(t);
}
```

I think this exact scenario was why we make `api.Graph` be `AutoClosable` in the first place :)